sematic/examples/lightning_resnet/checkpointing.py

```diff
@@ -0,0 +1,77 @@
+# Standard Library
+import io
+import logging
+from dataclasses import dataclass
+
+# Third-party
+import boto3
+import cloudpickle
+from pytorch_lightning.plugins.io.checkpoint_plugin import CheckpointIO
+
+# Sematic
+from sematic.types.types.aws.s3 import S3Location
+from sematic.utils.retry import retry
+
+logger = logging.getLogger(__name__)
+
+
+class SematicCheckpointIO(CheckpointIO):
```

Comment:
Why is this necessary? I read below that that `CheckpointIO` already works with S3.

Reply:
Because it only works with S3 *if* you use an extra third party dependency that I was unable to get to play nicely with our existing AWS dependencies. It seemed not worth it to me to add an extra, fickle depenency just to avoid this.